### PR TITLE
Improve StorageSwitch command to be more flexible

### PR DESCRIPTION
**Pull request by reneeb**
## Proposed change
The current Admin::Article::StorageSwitch command has hard coded values for target and source module.
If you have plugins installed that provide other backends, this command can't be used.

This pull requests make the command more flexible. It reads the installed backend module and allows
to switch to any of these backends from any backend.

## Type of change

- 🆙 - Dependency upgrade (e.g. libraries)
- 🐞 - Bugfix (non-breaking change which fixes an issue)
- 💎 - Code quality improvements to existing code or addition of unit tests
- [x] 🚀 - New feature (which adds functionality to an existing integration)

## Checklist

- [x] The code change is tested and works locally.(❗)
- [x] There is no commented out code in this PR.(❕)
- [x] You improved or added new unit tests.(❕/❗depends on the code)
- [x] Local ZnunyCodePolicy run passes successfully.(❗)
- [x] Local unit tests pass.(❗)
- [x] GitHub workflow ZnunyCodePolicy passes.(❗)
- [x] GitHub workflow unit tests pass.(❗)

### CODE REVIEW

**Comment by dennykorsukewitz** [comment]
@reneeb please check ZnunyCodePolicy.

```
[checked] Kernel/System/Console/Command/Admin/Article/StorageSwitch.pm
TidyAll::Plugin::OTRS::Perl::ObjectDependencies
The following objects are used in the code, but not declared as dependencies:
    'Kernel::System::Main',
```

**Comment by reneeb** [comment]
Thanks, will check this today...

File: Kernel/System/Console/Command/Admin/Article/StorageSwitch.pm [hunk 1]
```diff
@@ -25,13 +27,23 @@ our @ObjectDependencies = (
...
     my ( $Self, %Param ) = @_;
 
+    my @Backends     = $Self->_GetStorageBackends();
+    my $BackendRegex = join '|', @Backends;
+
     $Self->Description('Migrate article files from one storage backend to another on the fly.');
     $Self->AddOption(
         Name        => 'target',
-        Description => "Specify the target backend to migrate to (ArticleStorageDB|ArticleStorageFS).",
+        Description => "Specify the target backend to migrate to ($BackendRegex).",
```
Comment (jepf) [inline Kernel/System/Console/Command/Admin/Article/StorageSwitch.pm:37]: @reneeb 

The code looks like if you give option 'target', it still only supports ArticleStorageDB and ArticleStorageFS, not all found backends (you then get an error for the missing 'source' option).

Comment (reneeb) [reply Kernel/System/Console/Command/Admin/Article/StorageSwitch.pm:37]: No, it supports all backends found with _GetStorageBackends. You need to provide `--source` if you use a target other than the default storage backend. This is due to compatibility reasons.

Comment (jepf) [reply Kernel/System/Console/Command/Admin/Article/StorageSwitch.pm:37]: Thanks for the clarification.

**Pull request merged**